pkg/client/prometheus/servicemonitor.go

```diff
@@ -17,34 +17,82 @@ package prometheus
 import (
 	"context"
 	monv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
-	"github.com/kiegroup/kogito-cloud-operator/pkg/client"
+	monclientv1 "github.com/coreos/prometheus-operator/pkg/client/versioned/typed/monitoring/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientv1 "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 // ServiceMonitorInterface has functions that interact with Service Monitor instances in the Kubernetes cluster
 type ServiceMonitorInterface interface {
 	List(namespace string) (*monv1.ServiceMonitorList, error)
+	Create(ctx context.Context, serviceMonitor *monv1.ServiceMonitor, opts ...clientv1.CreateOption) error
+	Update(ctx context.Context, serviceMonitor *monv1.ServiceMonitor, opts ...clientv1.UpdateOption) error
+	Delete(ctx context.Context, serviceMonitor *monv1.ServiceMonitor, opts ...clientv1.DeleteOption) error
+	DeleteAllOf(ctx context.Context, serviceMonitor *monv1.ServiceMonitor, opts ...clientv1.DeleteAllOfOption) error
+	Patch(ctx context.Context, serviceMonitor *monv1.ServiceMonitor, patch clientv1.Patch, opts ...clientv1.PatchOption) error
 }
-
 type serviceMonitor struct {
-	client *client.Client
+	prometheusCli monclientv1.MonitoringV1Interface
 }
 
-func newServiceMonitor(c *client.Client) ServiceMonitorInterface {
-	client.MustEnsureClient(c)
+func newServiceMonitor(c monclientv1.MonitoringV1Interface) ServiceMonitorInterface {
 	return &serviceMonitor{
-		client: c,
+		prometheusCli: c,
 	}
 }
 
 func (s *serviceMonitor) List(namespace string) (*monv1.ServiceMonitorList, error) {
 	log.Debugf("List service monitor instances from namespace %s", namespace)
 
-	pList, err := s.client.PrometheusCli.ServiceMonitors(namespace).List(context.TODO(), metav1.ListOptions{})
+	pList, err := s.prometheusCli.ServiceMonitors(namespace).List(context.TODO(), metav1.ListOptions{})
 
 	if err != nil {
 		return nil, err
 	}
 
 	return pList, nil
 }
+
+// Create implements client.Client
+func (s *serviceMonitor) Create(ctx context.Context, serviceMonitor *monv1.ServiceMonitor, opts ...clientv1.CreateOption) error {
+	log.Debugf("Create service monitor instance : %s", serviceMonitor)
+	createOpts := &clientv1.CreateOptions{}
+	createOpts.ApplyOptions(opts)
```

Comment:
If I understand it correctly you copy the create options here to pass a copy to the Create command.
Is there any specific reason why original `opts` isn't passed instead?

Reply:
Original `opts` are slice of `CreateOption`(interface) but `ServiceMonitor` `Create` method expecting single value of `CreateOptions`(struct) that why I have to do that transformation